include/msgpack/v1/adaptor/map.hpp

```diff
@@ -48,7 +48,7 @@ namespace adaptor {
 template <typename K, typename V, typename Compare, typename Alloc>
 struct as<
     type::assoc_vector<K, V, Compare, Alloc>,
-    typename std::enable_if<msgpack::has_as<K>::value && msgpack::has_as<V>::value>::type> {
+    typename std::enable_if<msgpack::has_as<K>::value || msgpack::has_as<V>::value>::type> {
     type::assoc_vector<K, V, Compare, Alloc> operator()(msgpack::object const& o) const {
```

Comment:
@mdrohmann , I was wrong. I didn't understand what your fix mean at that time. Now, I understand it. Your fix is correct and simple :)